src/devices/Hcsr04/Hcsr04.cs

```diff
@@ -42,13 +42,39 @@ public Hcsr04(int triggerPin, int echoPin, PinNumberingScheme pinNumberingScheme
             _controller.OpenPin(_trigger, PinMode.Output);
 
             _controller.Write(_trigger, PinValue.Low);
+
```

Comment:
Not directly related to the bug, but mentioned in the issue as well: This ctor should have an overload that takes a GpioController instead of always creating its own.